base/util.jl

```diff
@@ -220,6 +220,45 @@ macro allocated(ex)
     end
 end
 
+function totalsizeof(x, ptr_cache = Set())
+    if in(pointer_from_objref(x), ptr_cache)
+        return 0
+    end
+    push!(ptr_cache, pointer_from_objref(x))
+
+    sz = 0
+    try
+        sz = sizeof(x)
```

Comment:
Defaulting to 0 in such cases is also a bit odd, but OTOH since symbols are reused perhaps that's as good as any other choice.

But if Symbols are the main way in which this fails, it might be better to say
```jl
if !isa(x, Symbol)
    try
        sz = sizeof(x)
...
```
because exceptions are a slow way of doing control flow. (More relevant for programmatic usage of `totalsize` than for usages of `whos()`, of course.) For reference, [the builtin definition of `sizeof`](https://github.com/JuliaLang/julia/blob/e97588db65f590d473e7fbbb127f30c01ea94995/src/builtins.c#L322-L346) might be helpful.